### PR TITLE
Fix encoder

### LINKED ISSUE #78: Encoding 0 bytes behavior
according to the kafka protocol doc: https://kafka.apache.org/protocol#protocol_types
"bytes" should be encoded as:

> Represents a raw sequence of bytes. First the length N is given as an INT32. Then N bytes follow.

But I can see that there is a special case in the library where it encode 0 bytes as `<<-1:32/signed-integer>>` [(Ref)](https://github.com/klarna/kafka_protocol/blob/master/src/kpro_lib.erl#L150), but not `[<<0:32>>, <<>>]` as what the doc says.

I am seeing the kafka broker rejecting a `sasl_authenticate` v0 request call with `auth_bytes` having 0 bytes content.

Is the special 0 bytes encoding behavior intended for some other case?

### PR DESCRIPTION
**Pull request by zmstone**
fixes #78

### CODE REVIEW

File: src/kpro_lib.erl [hunk 1]
```diff
@@ -137,6 +137,7 @@ encode(varint, I) when is_integer(I) -> kpro_varint:encode(I);
 encode(unsigned_varint, I) when is_integer(I) -> kpro_varint:encode_unsigned(I);
 encode(nullable_string, ?null) -> <<-1:16/?INT>>;
```
Comment (qzhuyan) [inline src/kpro_lib.erl:138]: What about this line? For backwards compatibility?

**Review by qzhuyan — APPROVED** [review]

File: src/kpro_lib.erl (hunk 1 above)
Comment (zmstone) [reply src/kpro_lib.erl:138]: no. `nullable_string` was invented for this purpose.
some `string` fields have the ambiguities of `null` and `""`, it's hard to know when to use `null` and when to use `""`.

**Comment by vikas15bhardwaj** [comment]
@zmstone Thanks for working on this issue. I tested with latest `:brod` and it pulled 4.0.1 `kafka_protocol` version. I noticed the change didn't pull as expected. I then checked release tagged 4.0.1 code and it seems tagged release is still having old code. I then checked `master` code branch which has the correct code. Can you check this. 


4.0.1  code https://github.com/kafka4beam/kafka_protocol/blob/94643e129c4063de67df75667ce3e22796a33bf2/src/kpro_lib.erl#L147

**Comment by zmstone** [comment]
Hi @vikas15bhardwaj 4.0.1 is old.
This fix is merged to master branch, but not released yet.

My plan was to include it in 4.1.0 
https://github.com/kafka4beam/kafka_protocol/pull/94

**Comment by zmstone** [comment]
Just release 4.0.2. also published to hex.

**Comment by vikas15bhardwaj** [comment]
Thanks @zmstone for publishing 4.0.2 release. My next problem is updating `brod` to use this release.

**Comment by vikas15bhardwaj** [comment]
@zmstone I was looking at updating `brod` to use `kafka_protocol` 4.0.2, I hope it's just a change in rebar.config file. Wondering the setup needed to compile/test. Seems no README on that. Can you help with it

**Comment by zmstone** [comment]
Hi @vikas15bhardwaj 

You are right, just change dependency version should be enough.
You can try to pin kafka_protocol version in your project for now.

**Pull request merged**